opentelemetry/proto/collector/dynamicconfig/v1/dynamicconfig_service.proto

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opentelemetry.proto.collector.dynamicconfig.v1;
+
+import "opentelemetry/proto/resource/v1/resource.proto";
+
+option java_multiple_files = true;
+option java_package = "io.opentelemetry.proto.collector.dynamicconfig.v1";
+option java_outer_classname = "DynamicConfigServiceProto";
+option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/dynamicconfig/v1";
+
+// DynamicConfig is a service that enables updating metric schedules, trace
+// parameters, and other configurations on an agent/SDK without having to
+// restart the instrumented application. The server itself is hosted on the
+// collector. In this way, the collector may act as a bridge between third-
+// party configuration services and the agent/sdk, piping updated configs
+// from a third-party source (or itself) to an instrumented application.
+service DynamicConfig {
+  rpc GetConfig (ConfigRequest) returns (ConfigResponse);
+}
+
+message ConfigRequest{
+  // Describes the version of the config service protocol to use. Required.
+  int32 proto_version = 1;
+
+  // The target resource. This resource pings the config server requesting
+  // updated configs, and the server responds with the config information.
+  opentelemetry.proto.resource.v1.Resource resource = 2;
+
+  // Optional. Identifies the last configuration fingerprint / version number.
+  // The fingerprint serves as a quick way for the config server to determine
+  // whether an update to the configs had occurred -- if the client request's
+  // fingerprint differs from the server's, then the server knows that a change
+  // had occured and an updated configuration should be sent.
+  int32 last_fingerprint = 3;
```

Comment:
Per below, I recommend changing this to a string.

Reply:
Changed